contentcuration/kolibri_public/utils/export_channel_to_kolibri_public.py

```diff
@@ -26,35 +62,30 @@ def export_channel_to_kolibri_public(
     # Note: The `categories` argument should be a _list_, NOT a _dict_.
     logger.info("Putting channel {} into kolibri_public".format(channel_id))
 
-    if channel_version is not None:
-        db_filename = "{id}-{version}.sqlite3".format(
-            id=channel_id, version=channel_version
-        )
+    versioned_db_filename = "{id}-{version}.sqlite3".format(
+        id=channel_id, version=channel_version
+    )
+    unversioned_db_filename = "{id}.sqlite3".format(id=channel_id)
+
+    versioned_db_storage_path = os.path.join(settings.DB_ROOT, versioned_db_filename)
+    unversioned_db_storage_path = os.path.join(
+        settings.DB_ROOT, unversioned_db_filename
+    )
+
+    if channel_version is None:
+        db_storage_path = unversioned_db_storage_path
     else:
-        db_filename = "{id}.sqlite3".format(id=channel_id)
-    db_location = os.path.join(settings.DB_ROOT, db_filename)
-
-    with storage.open(db_location) as storage_file:
-        with tempfile.NamedTemporaryFile(suffix=".sqlite3") as db_file:
-            shutil.copyfileobj(storage_file, db_file)
-            db_file.seek(0)
-            with using_content_database(db_file.name):
-                # Run migration to handle old content databases published prior to current fields being added.
-                call_command(
-                    "migrate",
-                    app_label=KolibriContentConfig.label,
-                    database=get_active_content_database(),
-                )
-                channel = ExportedChannelMetadata.objects.get(id=channel_id)
-                logger.info(
-                    "Found channel {} for id: {} mapping now".format(
-                        channel.name, channel_id
-                    )
-                )
-                mapper = ChannelMapper(
-                    channel=channel,
-                    public=public,
-                    categories=categories,
-                    countries=countries,
-                )
-                mapper.run()
+        db_storage_path = versioned_db_storage_path
+
+    with using_temp_migrated_database(db_storage_path):
```

Comment:
I'd rename it to something like `using_temp_migrated_content_database` to explicity declare that this is using the content database context.